File: RoomLiveDataSampleJava/app/src/main/java/com/websarva/wings/android/roomlivedatasamplejava/data/CocktailmemoRepository.java
package com.websarva.wings.android.roomlivedatasamplejava.data;

import android.app.Application;

import com.google.common.util.concurrent.ListenableFuture;
import com.websarva.wings.android.roomlivedatasamplejava.data.local.AppDatabase;
import com.websarva.wings.android.roomlivedatasamplejava.data.local.Cocktailmemo;
import com.websarva.wings.android.roomlivedatasamplejava.data.local.CocktailmemoDAO;

import java.util.List;
import java.util.concurrent.ExecutionException;

import androidx.lifecycle.LiveData;

public class CocktailmemoRepository {
	private AppDatabase _db;
	public CocktailmemoRepository(Application application) {
		_db = AppDatabase.getDatabase(application);
	}

	public LiveData<List<Cocktailmemo>> getCocktailmemoList() {
		CocktailmemoDAO cocktailmemoDAO = _db.createCocktailmemoDAO();
		return cocktailmemoDAO.findAll();
	}

	public Cocktailmemo getCocktailmemo(long cocktailId) {
		CocktailmemoDAO cocktailmemoDAO = _db.createCocktailmemoDAO();
		ListenableFuture<Cocktailmemo> future = cocktailmemoDAO.findByPK(cocktailId);
		Cocktailmemo cocktailmemo = null;
		try {
			cocktailmemo = future.get();
		}
		catch(ExecutionException ex) {
			ex.printStackTrace();
		}
		catch(InterruptedException ex) {
			ex.printStackTrace();
		}
		return cocktailmemo;
	}

	public boolean deleteInsertCocktailmemo(Cocktailmemo cocktailmemo) {
		CocktailmemoDAO cocktailmemoDAO = _db.createCocktailmemoDAO();
		ListenableFuture<Integer> futureDelete = cocktailmemoDAO.delete(cocktailmemo);
		boolean result = false;
		try {
			int resultDelete = futureDelete.get();
			if(resultDelete >= 0) {
				ListenableFuture<Long> futureInsert = cocktailmemoDAO.insert(cocktailmemo);
				long resultInsert = futureInsert.get();
				if(resultInsert >= 0) {
					result = true;
				}
			}
		}
		catch(ExecutionException ex) {
			ex.printStackTrace();
		}
		catch(InterruptedException ex) {
			ex.printStackTrace();
		}
		return result;
	}
}

File: RoomLiveDataSampleKotlin/app/src/main/java/com/websarva/wings/android/roomlivedatasamplekotlin/data/CocktailmemoRepository.kt
package com.websarva.wings.android.roomlivedatasamplekotlin.data

import android.app.Application
import androidx.lifecycle.LiveData
import com.websarva.wings.android.roomlivedatasamplekotlin.data.local.AppDatabase
import com.websarva.wings.android.roomlivedatasamplekotlin.data.local.Cocktailmemo

class CocktailmemoRepository(application: Application) {
	private val _db: AppDatabase

	init {
		_db = AppDatabase.getDatabase(application)
	}

	fun getCocktailmemoList(): LiveData<MutableList<Cocktailmemo>> {
		val cocktailmemoDAO = _db.createCocktailmemoDAO()
		return cocktailmemoDAO.findAll()
	}

	suspend fun getCocktailmemo(cocktailId: Long): Cocktailmemo? {
		val cocktailmemoDAO = _db.createCocktailmemoDAO()
		return cocktailmemoDAO.findByPK(cocktailId)
	}

	suspend fun deleteInsertCocktailmemo(cocktailmemo: Cocktailmemo): Boolean {
		val cocktailmemoDAO = _db.createCocktailmemoDAO()
		var result = false
		val resultDelete = cocktailmemoDAO.delete(cocktailmemo)
		if(resultDelete >= 0) {
			val resultInsert = cocktailmemoDAO.insert(cocktailmemo)
			if(resultInsert >= 0) {
				result = true
			}
		}
		return result
	}
}
